src/AppInitializer.tsx

```diff
@@ -130,6 +132,16 @@ function AppInitializer(
     SynapseClient.detectSSOCode();
   }, []);
 
+  useEffect(() => {
+    // SWC-6294: on mount, detect and attempt a client-side framebuster (mitigation only, easily bypassed by attacker)
+    if (window.top && window.top !== window) {
+      // If not sandboxed, make sure not to show any portal content (in case they block window unload via onbeforeunload)
+      setIsFramed(true);
+      // If sandboxed, this call will cause an uncaught js exception and portal will not load.
+      window.top.location = window.location;
+    }
+  }, []);
```

Comment:
Recommend putting this effect and the corresponding `useState` hook into one custom hook in a separate file